libs/shared/custom-events.ts

```diff
@@ -1,3 +1,12 @@
-export const orderChangeEventId = 'orderChangeEvent';
-export const cartQuantitiesChangedEventId = 'cartQuantitiesChangedEvent';
-export const sessionTokenReceivedEventId = 'sessionTokenReceivedEvent';
+export const SESSION_TOKEN_KEY = 'vendure-session-token';
+export const ORDER_CHANGE_EVENT = 'ORDER_CHANGE_EVENT';
+export const CART_QUANTITIES_CHANGED_EVENT = 'CART_QUANTITIES_CHANGED_EVENT';
+export const SESSION_TOKEN_RECEIVED_EVENT = 'SESSION_TOKEN_RECEIVED_EVENT';
+
+export const cartQuantitiesChangedEvent = (productVariantQuantities: Record<number, number>) =>
+	new CustomEvent(CART_QUANTITIES_CHANGED_EVENT, { detail: productVariantQuantities });
+
+export const sessionTokenReceivedEvent = (sessionToken: string) =>
+	new CustomEvent(SESSION_TOKEN_RECEIVED_EVENT, { detail: { sessionToken } });
+
+export const orderChangeEvent = () => new CustomEvent(ORDER_CHANGE_EVENT);
```

Comment:
All the events in the app are always all dispatched from the document, so I am wondering it it would make more sense to make these actually dispatch functions.

something like:
```suggestion
export const dispatchCartQuantitiesChangedEvent = (productVariantQuantities: Record<number, number>) =>
	document.dispatchEvent(new CustomEvent(CART_QUANTITIES_CHANGED_EVENT, { detail: productVariantQuantities }));

export const dispatchSessionTokenReceivedEvent = (sessionToken: string) =>
	document.dispatchEvent(new CustomEvent(SESSION_TOKEN_RECEIVED_EVENT, { detail: { sessionToken } }));

export const dispatchOrderChangeEvent = () => document.dispatchEvent(new CustomEvent(ORDER_CHANGE_EVENT));
```

so that then you can just call the functions and not having to repeat the `document.dispatchEvent(...)` every time
____

Or we could even have a single `dispatchEvent` method, something like this:
```ts
export const dispatchEvent = (event: DreamEvent, payload: any) => {
  let detail: any;
  switch(event) {
   case (DreamEvent.CART_QUANTITIES_CHANGED_EVENT):
     detail = ...;       // check that the payload is a Record<number, number> and create the detail object
     break;
   case (DreamEvent.....):
     .....
     break;
  }
  document.dispatchEvent(new CustomEvent(event, detail));
} 
```
where `DreamEvent` would be an string enum of something like that of the available events.

Reply:
I think this is a good point. I prefer the first implementation
_document.dispatchEvent_ inside every functions.
what do you think?

Reply:
Yes, the second version looks a bit uglier I think, I was suggesting it because I think it could be more convenient for development, you'd only need to remember/import a single function and intellisense should help you putting there the correct event name, I think :thinking: , anyways it is not a big deal....

....actually intellisense wouldn't help you with the payload anyways....

So maybe yeah it is better to just go with the first version :slightly_smiling_face::+1: